app/controllers/api/conversion_hosts_controller.rb

```diff
@@ -1,5 +1,96 @@
 module Api
   class ConversionHostsController < BaseController
     include Subcollections::Tags
+
+    # Create a conversion host and enable it. This operation will run as an
+    # MiqTask.
+    #
+    # Both the 'resource_type' and 'resource_id' are mandatory arguments,
+    # and the 'resource_type' must be either 'Host' or 'VmOrTemplate'.
+    #
+    # You may optionally pass in 'param_v2v_vddk_package_url' or 'auth_user'
+    # arguments as well.
+    #
+    # POST /api/conversion_hosts {
+    #   "name": "some_name",
+    #   "resource_type": "Host",
+    #   "resource_id": "7"
+    #   "param_v2v_vddk_package_url": "some_url"
+    #   "auth_user": "some_user"
+    # }
+    #
+    def create_resource(type, id, data)
+      raise BadRequestError, "resource_id must be specified" unless data['resource_id']
+      raise BadRequestError, "resource_type must be specified" unless data['resource_type']
+
+      resource_type = data['resource_type']
+      collection_type = resource_type.classify.constantize.table_name
```

Comment:
Should we work against a whitelist rather than constantizing whatever they send in?

Reply:
Hm, is `safe_constantize` good enough? Otherwise it looks like the intertubes agree that `constantize` is potentially unsafe.

Reply:
Right now nothing is preventing them from passing in `"resource_type": "MiqServer"` which we probably don't want.

Reply:
@djberg96 now I realize that if an unsupported resource is passed in, API won't fail it. It will fail only until queue_worker comes to work on it. So a whitelist is a better solution here.

(Somehow I had this impression that the `ConversionHost.enable_queue` will fail it if it is an unsupported resource, could be one of the prior implementations, i don't remember)

Reply:
I'm fine with adding the validation to the model if you prefer to do it there, but I think it needs to be done somewhere.

Reply:
Ok, I've added a whitelist to the controller, and it will now raise a bad request error if the resource type is invalid. I've added a spec to confirm as well.